src/main/scala/viper/gobra/translator/implementations/components/OptionImpl.scala

```diff
@@ -0,0 +1,220 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2020 ETH Zurich.
+
+package viper.gobra.translator.implementations.components
+
+import viper.gobra.translator.interfaces.Collector
+import viper.gobra.translator.interfaces.components.Options
+import viper.silver.{ast => vpr}
+
+class OptionImpl extends Options {
+  private val domainName : String = "Option"
+  private val typeVar : vpr.TypeVar = vpr.TypeVar("T")
+  private var generated : Boolean = false
```

Comment:
I like the var name in OptionToSeqImpl better:
```
  /**
     * Determines whether the Viper domain
     * should be generated upon finalisation.
     */
   private var generateDomain : Boolean = false
```

Reply:
I agree. Just updated it.